api/src/main/java/org/infinispan/api/async/AsyncCache.java

```diff
@@ -54,20 +64,60 @@
     */
    CompletionStage<Boolean> putIfAbsent(K key, V value);
```

Comment:
may I suggest to add default implementations to the interface:
```java
default CompletionStage<Boolean> putIfAbsent(K key, V value) {
     return putIfAbsent(key, value, CacheEntryExpiration.IMMORTAL);
}
```

Reply:
Done